coderd/promoauth/oauth2.go

```diff
@@ -214,10 +214,16 @@ func (c *Config) TokenSource(ctx context.Context, token *oauth2.Token) oauth2.To
 	return c.underlying.TokenSource(c.wrapClient(ctx, SourceTokenSource), token)
 }
 
+// InstrumentHTTPClient will always return a new http client with the same
+// transport as the one passed in.
 func (c *Config) InstrumentHTTPClient(hc *http.Client, source Oauth2Source) *http.Client {
-	// The new tripper will instrument every request made by the oauth2 client.
-	hc.Transport = newInstrumentedTripper(c, source, hc.Transport)
-	return hc
+	return &http.Client{
+		// The new tripper will instrument every request made by the oauth2 client.
+		Transport:     newInstrumentedTripper(c, source, hc.Transport),
```

Comment:
Why do we need to clone the provided transport in this func?

Reply:
You are right, I do not need to anymore. Before I had to because the default round tripped is a global, but now I'm making the client fresh, so it's all ok.